File: 240222/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.cpp
#include <iostream>

int main() {
    long long m, a, b;
    
    scanf("%d",&m);
    scanf("%d %d",&a, &b);

    int min_answer = 1e9;
    int max_answer = 0;
    int count;
    long long mid;
    long long s;
    long long e;

    for (long long i=a;i<=b;i++)
    {
        s = 1;
        e = m;
        count = 0;

        while (s<=e)
        {
            mid = (s+e)/2;
            count++;
            if (mid == i) break;
            else if(mid > i) e=mid-1;
            else s=mid+1;
        }
        if (count<min_answer) min_answer=count;
        if (count>max_answer) max_answer=count;
    }
    printf("%d %d",min_answer,max_answer);
    return 0;
}

File: 240222/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.py
m = int(input())
a, b = map(int, input().split())
min_answer = m
max_answer = 0
for i in range(a,b+1):
    s = 1
    e = m
    count = 0

    while s<=e:
        mid = (s+e)//2
        count += 1
        if mid == i:
            break
        elif mid > i:
            e = mid-1
        else:
            s = mid+1
    
    if min_answer>count:
        min_answer = count
    if max_answer<count:
        max_answer = count

print(min_answer, max_answer)